app/views/events/index.html.erb

```diff
@@ -41,9 +14,128 @@
+        "
+      >
+        <!-- Event Title -->
+        <h3 class="text-lg font-semibold text-gray-900 mb-2">
+          <div class="flex items-center space-x-2">
+            <!-- Bookmark icon -->
+            <%= render "bookmarks/editable_bookmark_icon", resource: event %>
+            <!-- Title link -->
+            <%= link_to "#{event.title}#{" [HIDDEN]" if current_user.super_user? && !event.publicly_visible}",
+            event_path(event),
+            class: "text-gray-900 hover:underline font-medium" %>
+          </div>
+        </h3>
+        <% if event.event_registrations.where(email: current_user.email).exists? %>
+          <span
+            class="
+              absolute top-2 right-2 text-xs bg-green-100 text-green-700 px-2 py-0.5
+              rounded-full
+            "
+          >
+            Registered
+          </span>
+          
+        <% end %>
+        <!-- Dates -->
+        <p class="text-sm text-gray-600 mb-1">
+          <span class="font-medium">Start:</span>
+          <%= event.start_date&.strftime("%B %d, %Y") %>
+        </p>
+        <p class="text-sm text-gray-600 mb-3">
+          <span class="font-medium">End:</span>
+          <%= event.end_date&.strftime("%B %d, %Y") %>
+        </p>
+
+        <div class="mt-3 flex gap-2">
+          <!-- Admin Edit -->
+          <% if current_user.super_user %>
+            <div class="mt-2">
+              <%= link_to "Edit", edit_event_path(event), class: "btn btn-secondary-outline" %>
+            </div>
+          <% end %>
+
+          <% if event.event_registrations.where(email: current_user.email).exists? %>
+          
+            <% registration = event.event_registrations.where(email: current_user.email).last %>
+            <div class="mt-2 ml-auto">
+              <%= link_to "De-register",
+              event_registration_path(registration),
+              class: "btn btn-secondary-outline",
+              method: :delete,
+              data: {
+                confirm: "Are you sure you want to de-register?",
+              } %>
+            </div>
+
+            
+          
+          <% elsif event.registerable? %>
+            <div class="mt-2 ml-auto">
+              <label class="flex items-center gap-2 cursor-pointer">
+                <%= check_box_tag "event_ids[]",
+                event.id,
+                false,
+                id: "event_ids_#{event.id}",
+                class:
+                  "event-checkbox h-5 w-5 text-blue-600 rounded border-gray-300 focus:ring-blue-500" %>
+                <span class="text-gray-700">Register</span>
+              </label>
+            </div>
+          <% else %>
+            <div class="mt-2 ml-auto">
+              <span class="text-sm text-gray-500 italic">Registration closed</span>
+            </div>
+          <% end %>
+        </div>
+         <!-- Add to Calendar -->
+          <% if event.event_registrations.where(email: current_user.email).exists? %>
```

Comment:
i think this needs to be:
`event.event_registrations.where(registrant: current_user).exists?`
bc event_registrations doesn't have an email field on it